### DISCUSSION TITLE
presence detect my own iPhone?

### DISCUSSION THREAD

**Discussion by pike2k** [h2zero/NimBLE-Arduino#729] [General]
Hello, new guy here. I have a project where I need to detect my own iPhone on an ESP32-S3. 
The idea was to use BLE, however I have read that iOS randomizes the bluetooth address? 
If this is true, that would shut down that plan pretty quickly...

**Comment by pike2k** [comment]
I will answer myself. Internet seems to agree that there's no point trying to detect the presence of an iPhone or Apple Watch.
Maybe a tag like 'Tile Mate' could be detected instead by ESP32-S3 + NimBLE? Any good example code out there for this task?
I am a hobbyist, not a real developer 👍

**Reply by DigiH** [reply]
Hi @pike2k 

Have a look at our iPhone, iPad and Apple Watch detection with [Theengs Gateway](https://github.com/theengs/gateway) (not implemented for an ESP32 with OpenMQTTGateway so far) and the [related paragraph on how to get the **Bluetooth Identity MAC Address**  and related **IRK** (Identity Resolving Key) for the devices.](https://gateway.theengs.io/use/use.html#getting-identity-resolving-key-irk-for-apple-watch-iphone-and-ipad)

Using [Theengs Decoder](https://github.com/theengs/decoder) we additionally implemented the unlocked state to be recognised, which allows for more security for a worn unlocked Apple Watch, and unlocked iPhone. Admittedly it not working for all iPhones as we haven't found the setting yet which makes it work reliably on some iPhones, but not on others, working reliably for all Apples Watches though.

Hope this helps in you also being able to track your iPhone and Apple Watch.

**P.S.:** ESPresense, using the same Identity MAC/IRK random MAC address resolving, should allow you to achieve the same directly on an ESP32, without having to port or write any code yourself.

**Reply by pike2k** [reply]
Thanks for the tip. I should have explained what I want to do with the presence detect. I have a project that will be installed in my car. It will do several things. The ESP32-S3 is also powering a built-in display (Device is a LilyGo T4-S3). I don't have many cpu cycles to spend on BLE and what I want to do is to limit functionality on the device unless (ideally) my iPhone is detected.

**Reply by DigiH** [reply]
You would just need to try and decrypt any received Bluetooth device's MAC address with the IRK, and if it resolves to your iPhone's Identity MAC it is it. 

Using a Tile or any other BLE beacon with a static MAC address will require the same scanning, depending on the frequency you want any device approaching to be detected and the cycles available, but with the disadvantage of static MAC address beacons being easily intercepted and then spoofed, while your iPhone's random MAC addresses makes it a lot harder and requiring ideal conditions to spoof.

I suppose it all comes down to how security/privacy relevant these several car functionalities are to you 😉

**Comment by pike2k** [comment]
Not sure if I should start a new discussion-thread or not. I've started experimenting with a BLE-tag from Xiaomi that presents itself like this when I perform a BLE-Scan:
Advertised Device: Name: iTAG, Address: ff:ff:11:d5:70:70, manufacturer data: 0501ffff11d570706602010300, serviceUUID: 0xffe0

The tag hopefully uses a static address (too early to be sure of this yet though).

I am looking for example-code that performs a one time scan (find and then stop scanning) for this specific address at startup. Which example is closest to my needs?
When I look at the existing examples nothing seems to come close to my needs.
 
Thanks in advance

**Comment by Staars** [comment]
Tracking of an iPhone is fully supported by Tasmota.

https://tasmota.github.io/docs/Bluetooth_MI32/#tracking-of-other-devices
